### PR TITLE
feat: ✅ generate test data for devstack

### PR DESCRIPTION
**Pull request by johnnagro**
## Description

Taking over from https://github.com/openedx/course-discovery/pull/3089 for @edx-abolger who is no longer with edX. Goal is to improve the provisioning tools to make it faster/easier to provision working enterprise setups.

## References

- [ENT-5355](https://openedx.atlassian.net/browse/ENT-5355)
- https://github.com/openedx/course-discovery/pull/3089
- https://openedx.atlassian.net/wiki/spaces/SOL/pages/3105718339/2021+Provision+Devstack+Enterprise+Data+Entirely+from+scripts

### CODE REVIEW

File: course_discovery/apps/course_metadata/management/commands/tests/test_create_test_courseruns.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from django.core.management import call_command
+from django.test import TestCase
+
+from course_discovery.apps.course_metadata.models import CourseRun, Organization
+from course_discovery.apps.course_metadata.tests.factories import PartnerFactory
+
+
+class CreateTestProgramCommandTests(TestCase):
```
Comment (waheedahmed) [inline course_discovery/apps/course_metadata/management/commands/tests/test_create_test_courseruns.py]: nit: `CreateTestCourseRunsCommandTests`

File: course_discovery/apps/course_metadata/management/commands/tests/test_create_test_courseruns.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    def setUp(self):
+        super().setUp()
+        self.partner = PartnerFactory()
+        self.command_args = [f'--partner_code={self.partner.short_code}']
+
+    def test_create_command(self):
```
Comment (waheedahmed) [inline course_discovery/apps/course_metadata/management/commands/tests/test_create_test_courseruns.py]: Same here, can we rename it to a more appropriate name? e.g. `test_create_course_runs_command`

**Review by waheedahmed — COMMENTED** [review]
Can we also add a few more test cases for the cases like updating the course API/marketing site URL etc? Otherwise looks good.

**Comment by johnnagro** [comment]
@waheedahmed there seems to be a flapping test, I don't think it is related to work my. It sometimes passes in Github CI, sometimes doesn't. I cannot reproduce locally. Any ideas?

`FAILED course_discovery/apps/course_metadata/tests/test_signals.py::TestCacheInvalidation::test_model_change`

**Comment by waheedahmed** [comment]
@johnnagro Yes, this test fails intermittently and isn't related to your changes. Please try rerunning the workflows.

**Review by waheedahmed — APPROVED** [review]

**Pull request merged**